### ISSUE TITLE
Add $set_error to filters within discount class

### ISSUE THREAD

**Issue by davidsherlock** [easydigitaldownloads/easy-digital-downloads#8752]
### Explain your enhancement (please be detailed)

For anyone extending via the available filters it would be useful to know if `set_error` is actually set on the specific check. Almost all the methods accept this, but none pass this variable in. This includes:

- is_product_requirements_met
- is_used
- is_valid
- is_active
- is_maxed_out
- is_min_price_met
- is_started

### Justification or use case

Easier third-party extension.

**Comment by ashleyfae** [comment]
I agree. Any filters on the output of a method should include all the parameters that went into said method.

**Comment by davidsherlock** [comment]
PR: #8760
